### PR TITLE
[integration-cli] fix race condition in kill tests

### LINKED ISSUE #28033: Flaky Test: DockerSuite.TestKillWithStopSignalWithDifferentSignalShouldKeepRestartPolicy
Seen failing on master, and z

```sh
 ----------------------------------------------------------------------
 FAIL: docker_cli_kill_test.go:77: DockerSuite.TestKillWithStopSignalWithDifferentSignalShouldKeepRestartPolicy
 
 docker_cli_kill_test.go:89:
     // Restart policy should still be in place, so it should be still running
     c.Assert(waitRun(cid), check.IsNil)
 ... value *errors.errorString = &errors.errorString{s:"condition \"\"false\" == \"true\"\" not true in time"} ("condition \"\"false\" == \"true\"\" not true in time")
 
 
 ----------------------------------------------------------------------
```

### PR DESCRIPTION
**Pull request by tophj-ibm**
Two commits because both are very closely related and will fail without the other.

The second commit should fix the Z CI, which was failing correctly because a test that tested container restart policy,
didn't have a restart policy. Then the first commit should fix the race condition that was causing the test to succeed on non-Z CI's in the first place.

commit 1.) Fixes a race condition in the kill tests where the container
would be killed but inspected before it's state changed. Fixes
this by waiting for a state change instead.

commit 2.) Fixes two tests that tested containers restart policy without
actually having a restart policy.

Fixes #28033 

Signed-off-by: Christopher Jones <tophj@linux.vnet.ibm.com>

race condition pig

### CODE REVIEW

**Comment by tophj-ibm** [comment]
ping @vdemeester 

Also can someone start the Z CI against this?

**Comment by tonistiigi** [comment]
LGTM

**Comment by coolljt0725** [comment]
LGTM

**Comment by coolljt0725** [comment]
`TestBuildNotVerboseFailureWithNonExistImage` test failed, but it's unrelated, re-trigger z

**Review by vdemeester — APPROVED** [review]
LGTM 🐸

**Comment by vdemeester** [comment]
Thanks a lot @tophj-ibm

File: integration-cli/docker_cli_kill_test.go [hunk 1]
```diff
@@ -63,12 +68,14 @@ func (s *DockerSuite) TestKillWithSignal(c *check.C) {
...
 	cid := strings.TrimSpace(out)
 	c.Assert(waitRun(cid), check.IsNil)
 
-	// Let's docker send a CONT signal to the container
+	// Let docker send a TERM signal to the container
+	// It will kill the process and disable the restart policy
 	dockerCmd(c, "kill", "-s", "TERM", cid)
+	c.Assert(waitExited(cid, 10*time.Second), check.IsNil)
 
 	out, _ = dockerCmd(c, "ps", "-q")
```
Comment (vdemeester) [inline integration-cli/docker_cli_kill_test.go]: We could and thus we would use `checker.Equals` instead of `checker.Contains` in the assert below 👼

Comment (tophj-ibm) [inline integration-cli/docker_cli_kill_test.go]: Yep, you can do that in a couple of the other tests in this file as well.

**Pull request merged**